src/fu-util.c

```diff
@@ -123,8 +123,12 @@ fu_util_update_device_changed_cb(FwupdClient *client, FwupdDevice *device, FuUti
 		priv->completion_flags |= FWUPD_DEVICE_FLAG_NEEDS_REBOOT;
 
 	/* same as last time, so ignore */
-	if (priv->current_device != NULL && fwupd_device_compare(priv->current_device, device) == 0)
+	if (priv->current_device == NULL ||
+	    g_strcmp0(fwupd_device_get_composite_id(priv->current_device),
+		      fwupd_device_get_composite_id(device)) == 0) {
+		g_set_object(&priv->current_device, device);
 		return;
+	}
```

Comment:
this same change is needed in `fu-tool.c`

Reply:
Good catch, fixed.